Sources/ArcGISToolkit/Components/FeatureFormView/FormInputs/Attachments/AttachmentCarouselItemThumbnail.swift

```diff
@@ -0,0 +1,112 @@
+    @State private var renameDialogueIsShowing = false
+    
+    /// The attachment data.
+    ///
+    /// Attachment data is provided asynchronously meaning there can be a short period where the data is
+    /// not present and we need to display alternative UI.
+    @State private var data: Data?
+    
+    /// Any error encountered while retrieving the attachment's data.
+    @State private var error: Error?
+    
+    /// The new name the user has provided for the attachment.
+    @State private var newAttachmentName = ""
+    
+    /// Creates UI to display a single form attachment.
+    /// - Parameter element: The form element containing the attachment.
+    /// - Parameter formAttachment: The attachment to be displayed.
+    init(element: AttachmentFormElement, formAttachment: FormAttachment) {
+        self.element = element
+        self.formAttachment = formAttachment
+    }
+    
+    var body: some View {
+        Group {
+            if !attachmentPreviewWasRequested {
+                Button {
+                    attachmentPreviewWasRequested = true
+                } label: {
+                    Image(systemName: "square.and.arrow.down")
+                        .font(.title3)
+                        .frame(maxWidth: .infinity, maxHeight: .infinity)
+                        .background(.thinMaterial)
+                }
+            } else if let data, let uiImage = UIImage(data: data) {
+                Image(uiImage: uiImage)
+                    .resizable()
+                    .aspectRatio(contentMode: .fill)
+                    .contextMenu {
+                        Button {
+                            newAttachmentName = formAttachment.name
+                            renameDialogueIsShowing = true
+                        } label: {
+                            Label("Rename", systemImage: "pencil")
+                        }
+                        Button(role: .destructive) {
+                            deletionWillStart = true
+                        } label: {
+                            Label("Delete", systemImage: "trash")
+                        }
+                    }
+            } else if data != nil || error != nil {
+                // Data was present but a UIImage wasn't initialized or an error
+                // was present.
+                Image(systemName: "exclamationmark")
+            } else {
+                // No data or error was present.
+                ProgressView()
+                    .progressViewStyle(CircularProgressViewStyle())
+            }
+        }
+        .alert("Rename attachment", isPresented: $renameDialogueIsShowing) {
+            TextField("New name", text: $newAttachmentName)
+            Button("Cancel", role: .cancel) { }
+            Button("Ok") {
+                formAttachment.setName(name: newAttachmentName)
+            }
+        }
+        .task(id: attachmentPreviewWasRequested) {
```

Comment:
Checking the "load" status here would be good, so if it's loaded, you can show the preview.